### PR TITLE
IBC-hooks: allow custom prefix in derivation of intermediate sender

### PR DESCRIPTION
**Pull request by pysel**
Closes: #5961

## What is the purpose of the change

Allow custom prefixes in derivation of intermediate sender

## Documentation and Release Note

changelog

### CODE REVIEW

**Comment by pysel** [comment]
devbot add changelog feat Allow providing custom bech32 and hash prefixes in derivation of intermediate sender in ibc-hooks

**Comment by pysel** [comment]
devbot merge base

File: x/ibc-hooks/client/cli/query.go [hunk 1]
```diff
@@ -42,26 +43,53 @@ func GetQueryCmd() *cobra.Command {
...
+				hashPrefix = types.SenderPrefix
+			} else {
+				hashPrefix = hashPrefixArg
+			}
+
+			var bech32Prefix string
+			if bech32PrefixArg == "" {
+				bech32Prefix = config.GetBech32AccountAddrPrefix()
+			} else {
+				bech32Prefix, err = getBech32CustomPrefix(config, bech32PrefixArg)
```
Comment (nicolaslara) [inline x/ibc-hooks/client/cli/query.go:86]: We can directly use the string supplied by the user as the prefix here. The idea is that you can use this to generate the derived addresses for other chains, so you can call `osmosisd q ibc-hooks wasm-hooks-sender --prefix=akash` and that would generate an akash-encoded address

Comment (pysel) [reply x/ibc-hooks/client/cli/query.go]: addressed

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -539,7 +539,7 @@ func (appKeepers *AppKeepers) WireICS20PreWasmKeeper(
 ) {
 	// Setup the ICS4Wrapper used by the hooks middleware
 	osmoPrefix := sdk.GetConfig().GetBech32AccountAddrPrefix()
-	wasmHooks := ibchooks.NewWasmHooks(hooksKeeper, nil, osmoPrefix) // The contract keeper needs to be set later
+	wasmHooks := ibchooks.NewWasmHooks(hooksKeeper, nil, osmoPrefix, ibchookstypes.SenderPrefix) // The contract keeper needs to be set later
```
[the remark this answers is not in the document]
Comment (pysel) [reply app/keepers/keepers.go]: hey @nicolaslara. [addressed](https://github.com/osmosis-labs/osmosis/pull/6003/commits/efd299047f0c9356f8f9d15a06e4df56133e7d74)

**Comment by pysel** [comment]
hey @nicolaslara. shall I update this PR?

**Pull request closed without merging**